# LuxOS: zdroj z gitu -> vydany archiv z download.gnome.org (sha256 overen proti gtk+-2.24.33.sha256sum
# od GNOME: ac2ac757f5942d318a311a54b0c80b5ef295f299c2a73c632f6bfb1ff49cc6da), "git apply" -> patch,
# bez .install (jen mazal cache v /usr/lib/gtk-2.0 - mimo adresare ovladacu; hook ji obnovi)
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -33,29 +33,28 @@
   shared-mime-info
 )
 makedepends=(
-  git
   glib2-devel
   gobject-introspection
   gtk-doc
 )
 source=(
-  "git+https://gitlab.gnome.org/GNOME/gtk.git#tag=$pkgver"
+  "https://download.gnome.org/sources/gtk+/2.24/gtk+-$pkgver.tar.xz"
   gtk-query-immodules-2.0.hook
   0001-Lower-severity-of-XID-collision-warnings.patch
   0002-Stop-looking-for-modules-in-cwd.patch
 )
-b2sums=('1b18d1cfef55466209cf93be45af15dc058a8b74d13ab590cfc7f0b09b0584adc62d4330aaed65185c0142cc8c326e4274c8e75e0af94bec5be3cfcca105c1e6'
+b2sums=('f3488b220aaee54bf36fa77e2112d060152e57938845d240014a7d43a5743b1d4de19cc09c683d9d238fd1967cd67cc3c493f2475e4e481f6709f5cdc652c65c'
         '9c531f9f605e1739e13c39c1cac22daddd9574f3082f18bcf0b9dfaa4c41f2485d55be03a9ed12fb4504d509f0d5ac63980a9d9349e3f80a06595c6430c78096'
         '45ecc976d9eb9d990fc204230aa052a6d1b2bdfdc94788be37d576ab262a1da49855eb46ecd4bfce4efde6e2f817a1660c6d1fa756be3b372f7f8d13b0ef0fd0'
         '06ca1c6f0e8f6a7c7a3cc08ce3d358af978d28fc4aa8d9e981883e3ad5adf7d821bcb27bc8b93bf65171a92396ac8f7ad62c90db501a492cca7c30b6081e957f')
 
 prepare() {
-  cd gtk
-  git apply -3 ../0001-Lower-severity-of-XID-collision-warnings.patch
+  cd gtk+-$pkgver
+  patch -Np1 -i ../0001-Lower-severity-of-XID-collision-warnings.patch
 
   # CVE-2024-6655: https://www.openwall.com/lists/oss-security/2024/09/09/1
   # https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/7361
-  git apply -3 ../0002-Stop-looking-for-modules-in-cwd.patch
+  patch -Np1 -i ../0002-Stop-looking-for-modules-in-cwd.patch
 
   sed -i '/AM_INIT_AUTOMAKE/s/]/ foreign]/' configure.ac
   autoreconf -fvi
@@ -72,7 +71,7 @@
 
   CFLAGS+=" -Wno-error=implicit-int -Wno-error=incompatible-pointer-types"
 
-  cd gtk
+  cd gtk+-$pkgver
   ./configure "${configure_options[@]}"
   sed -i -e 's/ -shared / -Wl,-O1,--as-needed\0/g' libtool
   make
@@ -90,9 +89,8 @@
     libgdk-x11-2.0.so
     libgtk-x11-2.0.so
   )
-  install=gtk2.install
 
-  make -C gtk DESTDIR="$pkgdir" install
+  make -C gtk+-$pkgver DESTDIR="$pkgdir" install
 
   install -Dm644 /dev/stdin "$pkgdir/usr/share/gtk-2.0/gtkrc" <<END
 gtk-icon-theme-name = "Adwaita"
--- a/gtk2.install
+++ /dev/null
@@ -1,3 +0,0 @@
-pre_remove() {
-    rm -f /usr/lib/gtk-2.0/2.10.0/immodules.cache
-}
